### PR TITLE
feat(moduletemplate): Wrap function module template wrappers with parens

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
N/A

**If relevant, link to documentation update:**

**Summary**
All browser engines currently can be signaled to lazy parse code by wrapping iife's with parents. After speaking with @fhinkel and a few other members of the V8 team, we were recommended that we use this heuristic technique. The change is very minimal in that in converts the output template to go from: 

```
/***/ function(module, exports, __webpack_require__) {
		var add = __webpack_require__(/*! ./math */ 1).add;
		exports.increment = function(val) {
			return add(val, 1);
		};
/***/ },
```

to 

```
/***/ (function(module, exports, __webpack_require__) {
		var add = __webpack_require__(/*! ./math */ 1).add;
		exports.increment = function(val) {
			return add(val, 1);
		};
/***/ }),
```

Overall findings by @nolanlawson state that load and performance gains of over 10%-50% depending on browser engine. https://github.com/nolanlawson/optimize-js#benchmark-overview

**Does this PR introduce a breaking change?**
No.

**Other information**

### CODE REVIEW

**Comment by nolanlawson** [comment]
Cool! Have you measured the impact of any bundle sizes to see what the load time improvement might be? Note that JS engines are changing all the time, and the impact can vary from codebase to codebase (even have a negative impact in some cases!), so it's good to actually be measuring this stuff.

In `optimize-js` I've been meaning to overhaul the benchmark system to provide better measurements, but if you'd like to use something yourself, I'd recommend [marky](https://github.com/nolanlawson/markymark) for adding `performance.mark`s and `measure`s and then just mark the time before inserting a `script` element into a page and then mark the time after. Curious to know what you find!

**Comment by citypaul** [comment]
Is there anywhere we can get an ELI5 (explain like I'm 5) explanation for this? I don't quite understand why this causes such a performance improvement?

**Comment by ev1stensberg** [comment]
@citypaul 

http://stackoverflow.com/questions/440739/what-do-parentheses-surrounding-a-object-function-class-declaration-mean

**Comment by gavinwahl** [comment]
@citypaul This description specifically addresses the performance improvement https://gist.github.com/getify/938c4a46013244cbd3cd750c0bc767ce

**Comment by TheLarkInn** [comment]
@nolanlawson, we ourselves don't have large enough example repos to warrent measurable results, we have had folks like @MikaAK who has used the separate plugin I wrote emulating this behavior. We would love to collect a list off large applications so we can do further perf analysis. Hard to get hands on in an easy way.

**Comment by citypaul** [comment]
@gavinwahl Thanks for that gist. That's exactly what I was looking for 👍

**Comment by statianzo** [comment]
> We would love to collect a list off large applications so we can do further perf analysis.

@TheLarkInn A large real-world repo with webpack is
https://github.com/instructure/canvas-lms

**Comment by TheLarkInn** [comment]
Thank you @statianzo! Have you tested out a build against webpack in master? I'm interesting to see the gains. Will probably test it myself anyways.

**Comment by statianzo** [comment]
@TheLarkInn I've been doing some internal tests at Domo with @aaronfrost using the `webpack-1` branch. I'll work on getting you some numbers next week.

**Comment by TheLarkInn** [comment]
Much appreciated!
